pkg/services/k8s/apiserver/entitystorage.go

```diff
@@ -447,6 +459,12 @@ func (s *entityStorage) GuaranteedUpdate(
 					return statusErr
 				}
 			}
+
+			// Hard to debug when we keep trying errors!
+			if true {
+				return err // return the real error
```

Comment:
I wish there was a statusError of type retryable, from Kubernetes, in which case, we will retry. Without that, we are left with individual error types we need to discern as worth retrying or not.